azurelinuxagent/ga/update.py

```diff
@@ -469,7 +489,10 @@ def _try_update_goal_state(self, protocol):
                 add_event(AGENT_NAME, op=WALAEventOperation.FetchGoalState, version=CURRENT_VERSION, is_success=True, message=message, log_event=False)
                 logger.info(message)
 
-            self._supports_fast_track = conf.get_enable_fast_track() and protocol.client.get_host_plugin().check_vm_settings_support()
+            try:
+                self._supports_fast_track = conf.get_enable_fast_track() and protocol.client.get_host_plugin().check_vm_settings_support()
+            except VmSettingsNotSupported:
```

Comment:
why only handle VmSettingsNotSupported exception, don't we see other exceptions here?

Reply:
We are trying to check whether the HostGAPlugin supports the vmSettings API. If it doesn't, it raises VmSettingsNotSupported. Any other exceptions are handled by the except clause on line 497.

Reply:
okay, what I thought was on vm migration scenario we may get VmSettingsSupportStopped exception but now I realized after looking at the code, this would be the second time call for fetch_vm_settings and first call was part of updating_goal_state. So, we handle that it first time and update the supports_fast_track to False so that it will raise VmSettingsNotSupported if we check again.